### ISSUE TITLE
Remove CNI configuration from the podman installation

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#11194]
Kubernetes has a problem to handle third-party packages using CNI.

Any installed config files will be used if present, in alphabetical order...
There is no way to select a specific config, especially one appearing later.

Since https://github.com/containers/podman/issues/2370

Podman now has a configuration option to select a different directory:

```toml
# The network table contains settings pertaining to the management of
# CNI plugins.

[network]

# Path to directory where CNI plugin binaries are located.
#
# cni_plugin_dirs = ["/usr/libexec/cni"]

# The network name of the default CNI network to attach pods to.
# default_network = "podman"

# Path to the directory where CNI configuration files are located.
#
# network_config_dir = "/etc/cni/net.d/"
```

`network.network_config_dir`

Changing this to a different directory, is the easiest way to fix kubeadm.

/etc/cni/net.d/87-podman-bridge.conflist

Another option would be to delete the file, and use `--network=host`.
But that would require any podman users to change, breaking some.

**Comment by afbjorklund** [comment]
This would avoid having to do workarounds like in PR #10384

**Comment by afbjorklund** [comment]
This also needs to be done for cri-o and for containerd packages.

**Comment by prezha** [comment]
> This also needs to be done for any cri-o and for containerd packages.
> 
> /etc/crictl.yaml
> /etc/cni/net.d/100-crio-bridge.conf
> /etc/cni/net.d/200-loopback.conf
> /etc/cni/net.d/10-containerd-net.conflist
> 
> See [#11184 (comment)](https://github.com/kubernetes/minikube/issues/11184#issuecomment-826046039)
> 
> The container runtime and the container network should be _configured_.
> 
> `criSocket`
> `--cri-socket`
> 
> Too bad that cniName can't be selected, only implicitly (lexicographic order)

crio and podman allow you to specify the cni name?

if _just_ the podman is the issue, can we then just rename the ***/etc/cni/net.d/87-podman-bridge.conflist*** to something like ***/etc/cni/net.d/999-podman-bridge.conflist*** (assuming 999 would always be the last one in lexicographic order)?

then, the `default_network = "podman"` from ***/etc/containers/containers.conf*** (as given above in https://github.com/kubernetes/minikube/issues/11194#issue-867088447) would still pick the right (in this example: /etc/cni/net.d/999-podman-bridge.conflist) config?

**Comment by afbjorklund** [comment]
podman is not the problem here, the network works fine. we just need to "hide" it from kubernetes.

**Comment by michaelhenkel** [comment]
as mentioned in https://github.com/kubernetes/minikube/pull/10384, cri-o has the same problem

**Comment by afbjorklund** [comment]
> as mentioned in #10384, cri-o has the same problem

That is true, but it is actually a container runtime (for Kubernetes) so it can theoretically keep the configuration **when selected**.

Podman doesn't run as a CRI, so we can move the Podman configuration away from the Kubernetes configuration for good...
For CRI-O (and containerd, soon docker) we need to make sure that they don't install any CNI configuration when not choosen.

Unfortunately it comes this way in the .deb packaging.

**Comment by afbjorklund** [comment]
> a thought: if _just_ the podman is the issue, can we then just rename the _**/etc/cni/net.d/87-podman-bridge.conflist**_ to something like _**/etc/cni/net.d/999-podman-bridge.conflist**_ (assuming 999 would always be the last one in lexicographic order)?

It doesn't matter _what_ we rename it to, since it's the only configuration file available it will always the first and the last.

The kindnet configuration (and some other CNI too, like flannel) is not created until _after_ the Kubernetes cluster has booted...

**Comment by mauilion** [comment]
This is also breaking expected functionality like "minikube start --cni=false"

In this configuration I would expect that no cni is configured. What happens instead is that the podman cni is used.

**Comment by afbjorklund** [comment]
> This is also breaking expected functionality like "minikube start --cni=false"
> 
> In this configuration I would expect that no cni is configured. What happens instead is that the podman cni is used.

Kubernetes doesn't have any feature to choose CNI, installing the Podman (or the cri-o or the containerd) package will break it...

This "alphabetical" configuration means that **no** other system components are able to use CNI, without changing config directory.
